.github/workflows/go.yml

```diff
@@ -24,8 +24,38 @@ jobs:
     - name: Build
       run: go build -v ./...
 
+    - name: Make coverage dirs
+      run: mkdir -p /tmp/unit_coverage /tmp/e2e_coverage
+
     - name: Test
-      run: go test -race -v ./...
+      run: go test -v -race -cover -coverpkg=./... $(go list ./... | grep -v /e2e/test/) -args -test.gocoverdir=/tmp/unit_coverage
+
+    - name: E2E Test
+      run: go test -v -cover -covermode=atomic -coverpkg=./... $(go list ./e2e/test/...) -args -test.gocoverdir=/tmp/e2e_coverage
```

Comment:
Running e2e with -race is slow, should we enable it?

Reply:
How much slower is it? I would lean towards yes - depending on how slow it is.

Reply:
2.145s -> 10.244s

Well, not really slow. And we can run e2e tests in parallel (each package is completely independent). let's use -race.